### ISSUE TITLE
[QUESTION]

### ISSUE THREAD

**Issue by smileWhileCoding** [NVIDIA/Megatron-LM#2124]
I’m trying train llama3 follow example/llama, when run the train, it seem stock at this step :"building train, validation, and test datasets for GPT ...", but gpu and cpu looks like working, i'm wondering what's happening here, and can i speed up it

**Comment by ericharper** [comment]
Hi, thanks for the issue. Please use the bug template and provide instructions to reproduce.

**Comment by ergang-bian** [comment]
> Hi, thanks for the issue. Please use the bug template and provide instructions to reproduce.

It finilly go to next step, takes one hour, and generate 217G data in data cache folder, i'm wondering is it a normal case.

[after model, optimizer, and learning rate scheduler are built] datetime: 2025-11-04 10:35:51 
> building train, validation, and test datasets ...
 > datasets target sizes (minimum size):
    train:      1953125000
    validation: 625000448
    test:       4096
> building train, validation, and test datasets for GPT ...
> finished creating GPT datasets ...
[after dataloaders are built] datetime: 2025-11-04 11:40:16


Input data is just 5K rows data from c4, and already be preprocessed to .bin and .idx. Train process is in a 8 H20 machine. use docker image nvcr.io/nvidia/pytorch:25.03-py3, in Megatron-LM main branch, same at core_r0.12.0 branch.
